flexmeasures/utils/plugin_utils.py

```diff
@@ -108,24 +100,75 @@ def register_plugins(app: Flask):
     sentry_sdk.set_context("plugins", app.config.get("LOADED_PLUGINS", {}))
 
 
-def check_config_settings(app, settings: dict):
-    """Make sure expected config settings exist."""
-    missing_optional_config_settings = []
-    if "optional" in settings:
-        for setting in settings["optional"]:
-            if app.config.get(setting) is None:
-                missing_optional_config_settings.append(setting)
-        if missing_optional_config_settings:
-            warnings.warn(
-                f"Missing optional config setting(s): {missing_optional_config_settings}",
-                category=ConfigurationWarning,
-            )
-    if "required" in settings:
-        missing_required_config_settings = []
-        for setting in settings["required"]:
-            if app.config.get(setting) is None:
-                missing_required_config_settings.append(setting)
-        if missing_required_config_settings:
-            raise ConfigurationError(
-                f"Missing required config setting(s): {missing_required_config_settings}"
-            )
+def check_config_settings(app, settings: Dict[str, dict]):
+    """Make sure expected config settings exist.
+
+    For example:
+
+        settings = {
+            "MY_PLUGIN_URL": {
+                "description": "URL used by my plugin for x.",
+                "level": "error",
+            },
+            "MY_PLUGIN_TOKEN": {
+                "description": "Token used by my plugin for y.",
+                "level": "warning",
+                "message": "Without this token, my plugin will not do y.",
+                "parse_as": str,
+            },
+            "MY_PLUGIN_COLOR": {
+                "description": "Color used to override the default plugin color.",
+                "level": "info",
+            },
+        }
+
+    """
+    assert isinstance(settings, dict), f"{settings} should be a dict"
```

Comment:
`settings` can be quite large, making it hard to read the logs. I'd print out the actual type, not the whole thing.